### ISSUE TITLE
🪲[Bug]: Override of PowerShellVersion is not taken into account

### ISSUE THREAD

**Issue by webalexeu** [PSModule/Build-PSModule#61]
### Describe the bug

The module build is not using the overrided (5.1) PowerShellVersion but is setting the default 7.4

### Version information

v3

### To reproduce

Define PowerShellVersion in manifest.psd1

### Code snippet

```powershell
@{
    # Minimum version of the PowerShell engine required by this module
    PowerShellVersion    = '5.1'
}
```

### Relevant output

```Shell
Link of the Action:
https://github.com/webalexeu/PSWinBGP/actions/runs/11684684422
```

**Comment by MariusStorhaug** [comment]
First off, big thank you for using the framework! Its awesome to see that it works well to run from a remote organization using the public reusable workflow and actions. Would be happy to discuss with you what you like with it and what you want to see improved on it.

I might loose you as a user of the framework here though; The behavior your are seeing is expected on v3 of the framework, the reasons can be found on the profile page of [PSModule](https://github.com/PSModule). All functionality that supported development of 5.1 should have been removed, if its not it will be at one point.

Leaving the issue open for a week in case you want to discuss this, I'm happy to hear if my assumptions are wrong regarding side-by-side installation of powershell on production servers or IT pro/developer machines.

**Comment by webalexeu** [comment]
Hello @MariusStorhaug ,

I was expecting to be able to still override this value if the module behind is working for both version (which is the case here)
I fully understand the direction to move to 7 but what will prevent the framework to just let the value to be overidden?

Thank you for your quick answer

**Comment by MariusStorhaug** [comment]
I agree that the override should still be possible. Let me have a look at how quickly I can re-add that.

**Comment by MariusStorhaug** [comment]
Deeper review of logs:
- I see that the override still is taken into consideration on Build-PSModule step. [Log link](https://github.com/webalexeu/PSWinBGP/actions/runs/11684684422/job/32536486070#step:4:3198)
- The issue seems to be in the test summary [Log link](https://github.com/webalexeu/PSWinBGP/actions/runs/11684684422/job/32536581676#step:5:217)
  - Root cause is likely here: <https://github.com/PSModule/Process-PSModule/blob/main/.github/workflows/workflow.yml#L388>
- Remains to be seen if this fixes it. The framework also touches the manifest in [Publish-PSModule](https://github.com/webalexeu/PSWinBGP/actions/runs/11684684422/job/32536659905#step:5:2376) to set the module version, but shouldn't touch the other metadata.
